cpp/daal/src/algorithms/dtrees/forest/classification/oneapi/df_classification_train_hist_oneapi_impl.i

```diff
@@ -294,9 +294,15 @@ services::Status ClassificationTrainBatchKernelOneAPI<algorithmFPType, hist>::co
 
         if (maxGroupBlocksNum > 1)
         {
-            const size_t nPartialHistograms = maxGroupBlocksNum > _minRowsBlocksForMaxPartHistNum ?
-                                                  _maxLocalHistograms / 2 :
-                                                  maxGroupBlocksNum < _maxLocalHistograms / 4 ? maxGroupBlocksNum : _maxLocalHistograms / 4;
+            size_t nPartialHistograms = maxGroupBlocksNum <= _minRowsBlocksForOneHist ? 1 : _maxLocalHistograms;
+
+            if (nPartialHistograms > 1 && maxGroupBlocksNum < _minRowsBlocksForMaxPartHistNum)
+            {
+                while (nPartialHistograms > 1 && nPartialHistograms * _minRowsBlocksForOneHist > maxGroupBlocksNum)
+                {
+                    nPartialHistograms >>= 1;
+                }
+            }
```

Comment:
method can be extracted

Reply:
I prefer to have it here for better logic understanding.